sklearn/cluster/_k_means_lloyd.pyx

```diff
@@ -27,14 +27,14 @@ np.import_array()
 
 
 def lloyd_iter_chunked_dense(
-        np.ndarray[floating, ndim=2, mode='c'] X,  # IN
-        floating[::1] sample_weight,               # IN
-        floating[::1] x_squared_norms,             # IN
-        floating[:, ::1] centers_old,              # IN
-        floating[:, ::1] centers_new,              # OUT
-        floating[::1] weight_in_clusters,          # OUT
-        int[::1] labels,                           # OUT
-        floating[::1] center_shift,                # OUT
+        floating[:, ::1] X,                # IN
```

Comment:
Without a `const floating`, this feels like the only way to document that the buffer is read-only:

```suggestion
        floating[:, ::1] X,                # IN READ-ONLY
```